### PR TITLE
CR-1110133: fixing multiple null checks around uuid usage

### PR DESCRIPTION
**Pull request by chvamshi-xilinx**
This change fixes the multiple uuid null checks.
1) xclbin uuid sysfs node comes empty when no xclbin is loaded. We were getting some uninitialized value earlier.
2) lock/unlock bistreams should fail whenever lock device called without loading xclbin
3) uuid constructor with string should initialize uuid with clear state.
4) don't open the context when xclbin is null

### CODE REVIEW

File: src/runtime_src/core/include/xrt/xrt_uuid.h [hunk 1]
```diff
@@ -71,12 +71,17 @@ class uuid
...
    *
    * A uuid string is 36 bytes with '-' at 8, 13, 18, and 23
    */
   explicit uuid(const std::string& uuid_str)
   {
+    if(uuid_str.empty())
+    {
+      uuid_clear(m_uuid);
+      return;
+    }
```
Comment (stsoe) [inline src/runtime_src/core/include/xrt/xrt_uuid.h:84]: Per convention space between if and brace, and brace on same line as condition.

**Review by stsoe — APPROVED** [review]
Minor style comment

**Comment by dayeh-xilinx** [comment]
retest this please. build failed for infra error.

File: src/runtime_src/core/edge/drm/zocl/zocl_sysfs.c [hunk 2]
```diff
@@ -72,7 +72,7 @@ static ssize_t xclbinid_show(struct device *dev,
 	struct drm_zocl_dev *zdev = dev_get_drvdata(dev);
 	ssize_t size;
 
-	if (!zdev)
+	if (!zdev || !zdev->zdev_xclbin || !zdev->zdev_xclbin->zx_uuid)
 		return 0;
```
Comment (mamin506) [inline src/runtime_src/core/edge/drm/zocl/zocl_sysfs.c:76]: If we print uuid_null, we don't really need any user space code change. Right?

Like this,
if (!zdev || !zdev->zdev_xclbin || !zdev->zdev_xclbin->zx_uuid) {
    size = sprintf(buf, "%pUb\n", &uuid_null);
} else {
    size = sprintf(buf, "%pUb\n", zdev->zdev_xclbin->zx_uuid);
}

**Review by mamin506 — APPROVED** [review]
The driver changes look good to me. If you don't want to make user space code change, please use my suggested code in zocl_sysfs.c.

File: src/runtime_src/core/edge/drm/zocl/zocl_sysfs.c (hunk 2 above)
Comment (chvamshi-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_sysfs.c:76]: Thats a good suggestion, but we still need to make user space changes to check whether uuid is null or not to call xclOpenConext. xrt_uuid constructor change is also required as it gets used in multiple places.

**Review by xdavidz — APPROVED** [review]

**Pull request merged**